tools/shared.py

```diff
@@ -772,6 +772,32 @@ def clean_temp():
     prepare_to_clean_temp(EMSCRIPTEN_TEMP_DIR) # this global var might change later
   return EMSCRIPTEN_TEMP_DIR
 
+class WarningManager:
+  WARN_ABSOLUTE_PATHS = False # by default, do not warn about absolute paths.
+  WARN_SEPARATE_ASM = True  # by default user is warned that they must load the separate 'asm.js' if html output is not used
+  WARN_ALMOST_ASM = True # disable this if you are expecting "almost asm" output, e.g. ALLOW_MEMORY_GROWTH=1 or SPLIT_MEMORY=1
+
+  @staticmethod
+  def capture_warnings(cmd_args):
+    assert cmd_args
+
+    for i in range(len(cmd_args)):
+      if cmd_args[i] == '-Wwarn-absolute-paths':
```

Comment:
Oh, I just realized that the absolute path flags were pre-existing. So I guess we maybe shouldn't change them. But the new ones can at least match the usual convention.